modules/container/src/main/java/org/mule/runtime/container/api/MuleInvocationHandler.java

```diff
@@ -65,11 +66,12 @@ protected ServiceInvocationHandler(Service service) {
    * See {@link InvocationHandler#invoke(Object, Method, Object[])}
    */
   protected Object doInvoke(Object proxy, Method method, Object[] args) throws Throwable {
-    if (isProxyClass(getService().getClass()) && getInvocationHandler(getService()) instanceof ServiceInvocationHandler) {
-      return ((ServiceInvocationHandler) getInvocationHandler(getService())).invoke(getService(), method, args);
+    if (isProxyClass(getProxiedObject().getClass())
+        && getInvocationHandler(getProxiedObject()) instanceof MuleInvocationHandler) {
```

Comment:
This condition seems to be duplicated on lines 48-49

Reply:
That is correct, already extracted into a method, thanks!